### PR TITLE
Added support for not using the built-in Authentication

### PR DESCRIPTION
**Pull request by BenniPlejd**
Currently there is no way of disabling the built-in Basic Authentication that the Parse-Dashboards uses. This causes issues if you want to use your own custom Authentication middleware instead.

I've added a new Configuration option named `noAuthentication`. If its value is set to `true`, then the built-in Authentication (~/Parse-Dashboard/Authentication.js) is not initialized.

### CODE REVIEW

**Comment by flovilmart** [comment]
@BenniPlejd I'd like to help on this one, but disabling authentication seems be be a bad idea, I'd rather have a new authentication provider, compatible with passport, instead of disabling it altogether. Then you're free to have security holes on your dashboard, but at least, we've provided a best effort to prevent users from inadvertently expose their dashboard. What do you think?

**Comment by flovilmart** [comment]
Also, if you want to disable your authentication, and use your own middleware, just don't specify any authentication info in it, that would be easier I believe no?

**Comment by romanbsd** [comment]
We used a bit different approach:
https://github.com/parse-community/parse-dashboard/issues/751

**Comment by yamitzky** [comment]
I agree with this PR. I am using the parse dashboard in a private network, so it must be completely secure without any authentication.

**Comment by yamitzky** [comment]
@flovilmart I agree with this PR. I am using the parse dashboard in a private network, so it must be completely secure without any authentication.

File: README.md [hunk 1]
```diff
@@ -147,6 +147,8 @@ Parse Dashboard supports adding an optional icon for each app, so you can identi
 
 You can set `appNameForURL` in the config file for each app to control the url of your app within the dashboard. This can make it easier to use bookmarks or share links on your dashboard.
 
+If you wish to use a custom authentication system, then you can disable Parse Dashboard's built-in authentication system by setting `noAuthentication` to `false`. The default value is false if not specified.
```
Comment (acinader) [inline README.md]: i think this is backward.  

"you can disable Parse Dashboard's built-in authentication system by setting `noAuthentication` to `false`. "

should be true, no?

**Comment by acinader** [comment]
While I defer to your judgment, my $.02 would be that an option to turn off authentication is not a bad idea.  I can imagine a few legitimate use cases: running on your local box or in a VPN environment.

The distinction between this and allowing email to be returned in an unauthenticated rest call is that I don't think that a lazy developer would use this feature just to avoid doing some work.

**Comment by flovilmart** [comment]
If you don’t provide any users, there’s no auth...

**Comment by acinader** [comment]
oh.  nevermind.

**Pull request closed without merging**